Source/Public/Get-ADOPSServiceConnection.ps1

```diff
@@ -1,42 +1,43 @@
 function Get-ADOPSServiceConnection {
     [CmdletBinding()]
     param (
-        [Parameter()]
-        [ValidateNotNullOrEmpty()]
-        [string]$Name,
-
         [Parameter(Mandatory)]
         [ValidateNotNullOrEmpty()]
         [string]$Project,
 
         [Parameter()]
         [ValidateNotNullOrEmpty()]
-        [string]$Organization
+        [string]$Name,
+
+        [Parameter()]
+        [ValidateNotNullOrEmpty()]
+        [string]$Organization,
+        
+        [Parameter()]
+        [switch]
+        $IncludeFailed
     )
     
     # If user didn't specify org, get it from saved context
     if ([string]::IsNullOrEmpty($Organization)) {
         $Organization = GetADOPSDefaultOrganization
     }
 
-    $Uri = "https://dev.azure.com/$Organization/$Project/_apis/serviceendpoint/endpoints?api-version=7.1-preview.4"
+    $Uri = "https://dev.azure.com/$Organization/$Project/_apis/serviceendpoint/endpoints?includeFailed=$IncludeFailed&api-version=7.1-preview.4"
     
     $InvokeSplat = @{
-        Method       = 'Get'
-        Uri          = $URI
+        Method = 'Get'
+        Uri    = $URI
     }
 
-    $AllPipelines = (InvokeADOPSRestMethod @InvokeSplat).value
+    $ServiceConnections = (InvokeADOPSRestMethod @InvokeSplat).value
 
     if ($PSBoundParameters.ContainsKey('Name')) {
-        $Pipelines = $AllPipelines | Where-Object {$_.name -eq $Name}
-        if (-not $Pipelines) {
```

Comment:
Is there a reason to remove the 'throw' message here? If a name is given and no name is matched I think we should throw an error instead of returning nothing. Removing this error would possibly be a breaking change also as people may trust the error in scripts.

Reply:
Considering the design in other get functions - none of them throws and error, instead they return an empty response such as:
- https://github.com/AZDOPS/AZDOPS/blob/main/Source/Public/Get-ADOPSGroup.ps1
- https://github.com/AZDOPS/AZDOPS/blob/main/Source/Public/Get-ADOPSNode.ps1
- https://github.com/AZDOPS/AZDOPS/blob/main/Source/Public/Get-ADOPSProject.ps1
- https://github.com/AZDOPS/AZDOPS/blob/main/Source/Public/Get-ADOPSRepository.ps1
- https://github.com/AZDOPS/AZDOPS/blob/main/Source/Public/Get-ADOPSVariableGroup.ps1

Perhaps it's more of a design principal when it comes to the module, but I thought I'd make it match what the majority of the other get functions has (There's only 1 other that uses throw when it does not return a value).
It's; https://github.com/AZDOPS/AZDOPS/blob/main/Source/Public/Get-ADOPSPipeline.ps1

Reply:
Well noted, and really good point. Please make an issue and report it and we will pin it to version 3! Approved.